splash/pages/pages_service.py

```diff
@@ -41,7 +41,7 @@ def retrieve_by_page_type(self,
         query = {'page_type': page_type}
         return self.retrieve_multiple(current_user, page, query, page_size)
 
-    def update(self, current_user: User, data: Page, uid: str):
+    def update(self, current_user: User, data: NewPage, uid: str):
```

Comment:
In our awkward API, wouldn't this be a Page and not a NewPage?

FWIW, I'm thinking of dumping the whole NewX thing in favor of just making uid optional.

Reply:
No, it would be NewPage because Page includes all this extra metadata which we don't want the user changing. The metadata should only be writeable by our own service,

Reply:
In that case...why not a third model (blech)? Updating a NewPage seems odd.

Reply:
I mean I could make third model but it would be the exact same as NewPage.